include/execution.hpp

```diff
@@ -2889,227 +2709,202 @@ namespace std::execution {
+          requires sender_to<__member_t<_Self, _Sender>, _Receiver>
+        friend auto tag_invoke(connect_t, _Self&& __self, _Receiver&& __rcvr)
+            -> __operation1<_SchedulerId, __x<__member_t<_Self, _Sender>>, __x<decay_t<_Receiver>>> {
+          return {__self.__sched_, ((_Self&&) __self).__sndr_, (_Receiver&&) __rcvr};
+        }
 
-          template <__decays_to<__sender> _Self, receiver _Receiver>
-            requires sender_to<__member_t<_Self, _Sender>, _Receiver>
-          friend auto tag_invoke(connect_t, _Self&& __self, _Receiver&& __rcvr)
-              -> __operation1<_SchedulerId, __x<__member_t<_Self, _Sender>>, __x<decay_t<_Receiver>>> {
-            return {__self.__sched_, ((_Self&&) __self).__sndr_, (_Receiver&&) __rcvr};
-          }
+        template <__one_of<set_value_t, set_stopped_t> _Tag>
+        friend _Scheduler tag_invoke(get_completion_scheduler_t<_Tag>, const __sender& __self) noexcept {
+          return __self.__sched_;
+        }
 
-          template <__one_of<set_value_t, set_stopped_t> _Tag>
-          friend _Scheduler tag_invoke(get_completion_scheduler_t<_Tag>, const __sender& __self) noexcept {
-            return __self.__sched_;
-          }
+        template <__sender_queries::__sender_query _Tag, class... _As>
+          requires __callable<_Tag, const _Sender&, _As...>
+        friend auto tag_invoke(_Tag __tag, const __sender& __self, _As&&... __as)
+          noexcept(__nothrow_callable<_Tag, const _Sender&, _As...>)
+          -> __call_result_if_t<__sender_queries::__sender_query<_Tag>, _Tag, const _Sender&, _As...> {
+          return ((_Tag&&) __tag)(__self.__sndr_, (_As&&) __as...);
+        }
 
-          template <__sender_queries::__sender_query _Tag, class... _As>
-            requires __callable<_Tag, const _Sender&, _As...>
-          friend auto tag_invoke(_Tag __tag, const __sender& __self, _As&&... __as)
-            noexcept(__nothrow_callable<_Tag, const _Sender&, _As...>)
-            -> __call_result_if_t<__sender_queries::__sender_query<_Tag>, _Tag, const _Sender&, _As...> {
-            return ((_Tag&&) __tag)(__self.__sndr_, (_As&&) __as...);
-          }
+        template <class...>
+          using __value_t = completion_signatures<>;
 
-          template <__decays_to<__sender> _Self, class _Env>
-            friend auto tag_invoke(get_completion_signatures_t, _Self&&, _Env)
-              -> completion_signatures_of_t<__member_t<_Self, _Sender>, _Env>;
-        };
-    } // namespace __impl
+        template <__decays_to<__sender> _Self, class _Env>
+          friend auto tag_invoke(get_completion_signatures_t, _Self&&, _Env) ->
+            make_completion_signatures<
+              __member_t<_Self, _Sender>,
+              _Env,
+              make_completion_signatures<
+                schedule_result_t<_Scheduler>,
+                _Env,
+                completion_signatures<set_error_t(exception_ptr)>,
```

Comment:
General readability note that is impeding my ability to review the code:

It is very hard for me to tell which facility this is even involved with.  There's __sender, __operation1, __receiver1, and other structs preceding this line.  My main hint as to what facility this belongs to is a comment a couple hundred lines up though, where there's a mention of execution.senders.adaptors.schedule_from.

My suggestion is to add more redundancy in naming.  Instead of __sender, use __sender_schedule_from.  The extra redundancy can help readers like myself.  It might even help diagnostics a bit.

Reply:
I get where you're coming from. All these types are in an appropriately named namespace, but the namespace is often far from where the sender/receiver types are defined, so it might be better to do as you suggest.